src/ring_theory/hahn_series.lean

```diff
@@ -157,12 +165,28 @@ instance : distrib_mul_action R (hahn_series Γ V) :=
 
 end distrib_mul_action
 
-instance [linear_order Γ] {V : Type*} [semiring R] [add_comm_monoid V] [semimodule R V] :
+section scalar
+
+variables [linear_order Γ] [semiring R] {V : Type*} [add_comm_monoid V] [semimodule R V]
+
+instance :
   semimodule R (hahn_series Γ V) :=
 { zero_smul := λ _, by { ext, simp },
   add_smul := λ _ _ _, by { ext, simp [add_smul] },
   .. hahn_series.distrib_mul_action }
 
+variables {S : Type*} [semiring S] [semimodule S V]
+
+instance [semimodule R S] [is_scalar_tower R S V] :
```

Comment:
I think you can downgrade semimodule to has_scalar

Reply:
`smul_zero` is necessary for the product to be a Hahn series.